test/e2e/network_test.go

```diff
@@ -656,17 +657,16 @@ func (tc *testContext) createWindowsServerJob(name, pwshCommand string, affinity
 	windowsOS := &v1.PodOS{Name: v1.Windows}
 	windowsServerImage := tc.getWindowsServerContainerImage()
 	command := []string{powerShellExe, "-command", pwshCommand}
-	return tc.createJob(name, windowsServerImage, command, windowsNodeSelector, windowsTolerations, affinity, windowsOS)
+	return tc.createJob(name, true, windowsServerImage, "", command, windowsNodeSelector,
+		windowsTolerations, affinity, windowsOS)
 }
 
 // createJob creates a job on the cluster using the given parameters
-func (tc *testContext) createJob(name, image string, command []string, selector map[string]string,
-	tolerations []v1.Toleration, affinity *v1.Affinity, os *v1.PodOS) (*batchv1.Job, error) {
+func (tc *testContext) createJob(name string, nameGenerated bool, image, serviceAccountName string, command []string,
```

Comment:
I'm not convinced this is necessary.
Adding more and more bools to all our test setup functions make them harder to read.

Reply:
+1 to removing the fixed name introduction if possible

Reply:
I'm fine reverting it. Went with this to avoid repeating the logic for job definition and creation.